marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/WriteBatcherImpl.java

```diff
@@ -1331,4 +1335,20 @@ public boolean awaitCompletion(long timeout, TimeUnit unit) throws InterruptedEx
       return tasks;
     }
   }
+
+
+@Override
+public WriteBatcher withDefaultMetadata(DocumentMetadataHandle handle) {
+    this.docMetadataHandle = handle;
+    return this;
+}

```

Comment:
We'll also need to find the place in the WriteBatcher implementation where the added DocumentWriteOperations are batched for the server and emit the defaultMetadata as the first payload.

Finding the right place may require reading through the implementation and / or stepping through the implementation in a debugger.